### ISSUE TITLE
Evaluation with a PyTorch model from another framework?

### ISSUE THREAD

**Issue by Petros626** [tudelft-iv/view-of-delft-dataset#113]
Hello guys,

awesome work!

Is it possible to use a `model.pt` from my framework and evaluate it with my predictions and gt-labels?

**Comment by andraspalffy** [comment]
If you run your inference in your own framework, you can export the detections. Once you convert that into the format of the VoD dataset (very similar to KITTI) then you can evaluate them.

**Comment by Petros626** [comment]
When I provide for the detections only rotated 2D BBoxes from the BEV perspective, how can the metrics for bbox(2D I guess) , 3D calculated be computed? 

I believe that then only bev AP would be possible,instead of all metrics.

**Comment by andraspalffy** [comment]
Without knowing the specifics of your setup, I think you are right, BEV AP is the only reasonable one. You can try 3D by assigning a "fake" height to all boxes (e.g. bike = 1.7 m tall) and vertical offset (e.g. every box is on the plane = -1 m or so), but will not be very meaningful.

**Comment by Petros626** [comment]
Thanks for the clarification Andras, coming back to you in case of further issues.

**Comment by Petros626** [comment]
Hey @andraspalffy,

I'm trying now to validate my bird's eye view dataset based on `KITTI` (see attachment). For the training framework I was forced to convert x, y, z and h, w, l (in m) to 640x640 bev-image space and further to normalize the coordinates [0...1].

![Image](https://github.com/user-attachments/assets/7e0870ec-5f0e-4a2e-80f4-d4822d80da8d)

How can I reverse this procedure for my gt-data and the saved predictions? 
From my understanding of your evaluation code every parameter of the box is in the camera frame (KITTI format), which would lead to this:
BEV label (normalized) -> BEV label (pixel space) -> BEV label LiDAR (only x,y,l,w,r) -> BEV label Camera (x,y,l,w,r)


Update 02.05.2025:
The mentioned pipeline worked to obtain very precise (exceptions objects at the edge of the bev picture) the original gt-labels from the KITTI training data.
